lambdas/data-migration2/src/storeErrors.ts

```diff
@@ -0,0 +1,63 @@
+import { WriteStream } from 'node:fs';
+
+const fs = require('fs');
+const moment = require('moment');
+const { finished } = require('stream');
+const { promisify } = require('util');
+const { s3 } = require('@cumulus/aws-client/services');
+
+/**
+ * Helper to create error file write stream
+ * @param {string} migrationName         - Name of migration
+ * @param {string | undefined} timestamp - Timestamp for unit testing
+ * @returns {Object}                     - Object containing error write stream and file path string
+ */
+export const createErrorFileWriteStream = (migrationName: string, timestamp?: string) => {
+  const dateString = timestamp || moment.utc().format('YYYY-MM-DD_HH:MM:SS.SSSS');
+  const filepath = `${migrationName}ErrorLog-${dateString}.json`;
+  const errorFileWriteStream = fs.createWriteStream(filepath);
+  errorFileWriteStream.write('{ "errors": [\n');
+
+  return { errorFileWriteStream, filepath };
+};
+
+/**
+ * Helper to close error file write stream
+ * @param {WriteStream} errorFileWriteStream - Error file write stream to close
+ * @returns {Promise<void>}
+ */
+export const closeErrorFileWriteStream = async (errorFileWriteStream: WriteStream) => {
+  errorFileWriteStream.end('\n]}');
+  const asyncFinished = promisify(finished);
+  await asyncFinished(errorFileWriteStream);
+};
+
+/**
+ * Store migration errors JSON file on S3.
+ *
+ * @param {Object} params
+ * @param {string} params.bucket                - Name of S3 bucket where file will be uploaded
+ * @param {string} params.filepath              - Write Stream file path
+ * @param {string} params.migrationName         - Name of migration
+ * @param {string} params.stackName             - User stack name/prefix
+ * @param {string | undefined} params.timestamp - Timestamp for unit testing
+ * @returns {Promise<void>}
+ */
+export const storeErrors = async (params: {
+  bucket: string,
+  filepath: string,
+  migrationName: string,
+  stackName: string,
+  timestamp?: string,
+}) => {
+  const { bucket, filepath, migrationName, stackName, timestamp } = params;
+  const fileKey = `data-migration2-${migrationName}-errors`;
+  const dateString = timestamp || moment.utc().format('YYYY-MM-DD_HH:MM:SS');
```

Comment:
Can we avoid using moment and introducing the dependency by doing `new Date().toISOString()`?

Reply:
It also might be nice for this function to log the name of the object it wrote to S3 for traceability